integrations/terraform/go.mod

```diff
@@ -12,8 +12,8 @@ require (
 	github.com/gravitational/teleport/api v0.0.0 // replaced
 	github.com/gravitational/trace v1.4.0
 	github.com/hashicorp/terraform-plugin-framework v0.10.0
-	github.com/hashicorp/terraform-plugin-go v0.12.0
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.10.1
+	github.com/hashicorp/terraform-plugin-go v0.22.0
```

Comment:
This is a big jump. WDYT, @hugoShaka ?

Reply:
I'm not affraid about the plugin-sdk, I will need to test the plugin-go update manually.

If the integrations unit tests are passing it's a good sign though

Reply:
Thanks. Let us know when you are happy to move ahead.

Reply:
I managed to bump up to 0.18.0 without breaking. anything, but 0.19 requires upgrading `github.com/hashicorp/terraform-plugin-framework` which is not possible today

Reply:
Fair enough. Do you wanna mail your 0.18.0 PR? Shall we close this one?

Reply:
I'm preparing a commit for this and also configuring a hold on the package. I'd like tot try out a few things to see if there are low hanging fruits allowing us to bump a couple versions higher